### PR TITLE
[14.0][FIX] stock_picking_cancel_confirm: cancel pickings without lines

### PR DESCRIPTION
**Pull request by ps-tubtim**
This pull request fix is about canceled pickings without lines.

**Before:**  Pickings not cancelled
cancel_picking

**After:** Pickings cancelled
cancel_picking_2

### CODE REVIEW

File: stock_picking_cancel_confirm/wizard/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import cancel_confirm
```
Comment (kittiu) [inline stock_picking_cancel_confirm/wizard/__init__.py:3]: **In my option, we don't need to patch any of this wizard.**
It was the logic of Odoo core in picking.action_cancel() that do not cancel picking (whether it is a bug or the intention)

BUT, if we need to fix it we should fix at the right place, which is in action_cancel() itself.

Comment (kittiu) [inline stock_picking_cancel_confirm/wizard/__init__.py]: **In my option, we don't need to patch any of this wizard.**
It was the logic of Odoo core in picking.action_cancel() that do not cancel picking (whether it is a bug or the intention)

BUT, if we need to fix it we should fix at the right place, which is in action_cancel() itself.

Comment (kittiu) [reply stock_picking_cancel_confirm/wizard/__init__.py]: I propose the following code for **def action_cancel()**

```
    def action_cancel(self):
        if not self.filtered("cancel_confirm"):
            return self.open_cancel_confirm_wizard()
        res = super().action_cancel()
        # Ensure to cancel the picking w/o moves that wasn't cancelled by odoo core
        self.filtered(lambda l: l.state != "cancel" and not l.move_lines).write({"state": "cancel"})
        return res
```

Comment (ps-tubtim) [reply stock_picking_cancel_confirm/wizard/__init__.py]: @kittiu Thank you for your suggestion. I will change it.

File: stock_picking_cancel_confirm/models/stock_picking.py [hunk 2]
```diff
@@ -12,4 +12,9 @@ class StockPicking(models.Model):
     def action_cancel(self):
         if not self.filtered("cancel_confirm"):
             return self.open_cancel_confirm_wizard()
-        return super().action_cancel()
+        res = super().action_cancel()
+        # Ensure to cancel the picking w/o moves that wasn't cancelled by odoo core
+        self.filtered(lambda l: l.state != "cancel" and not l.move_lines).write(
```
Comment (rousseldenis) [inline stock_picking_cancel_confirm/models/stock_picking.py:17]: @ps-tubtim Could you add a test with no moves ?

But I think as state field is computed one and readonly, this should not work.

In core, picking remains in 'draft' if there is no move inside.

**Comment by rousseldenis** [comment]
@ps-tubtim

**Pull request closed without merging**